File: nettest-shared/src/main/java/at/alladin/nettest/shared/berec/collector/api/v1/dto/shared/RttInfoDto.java
package at.alladin.nettest.shared.berec.collector.api.v1.dto.shared;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Contains round trip time information measured during the measurement on the measurement agent.
 * 
 * @author alladin-IT GmbH (lb@alladin.at)
 * @author alladin-IT GmbH (bp@alladin.at)
 *
 */
@io.swagger.annotations.ApiModel(description = "Contains round trip time information measured during the measurement on the measurement agent.")
@JsonClassDescription("Contains round trip time information measured during the measurement on the measurement agent.")
public class RttInfoDto {

	/**
	 * List of all measured RTTs.
	 */
	@io.swagger.annotations.ApiModelProperty("List of all measured RTTs.")
	@JsonPropertyDescription("List of all measured RTTs.")
	@Expose
	@SerializedName("rtts")
	@JsonProperty("rtts")
	private List<RttDto> rtts;

	/**
	 * The number of RTT packets to send, as instructed by the server.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The number of RTT packets to send, as instructed by the server.")
	@JsonPropertyDescription("The number of RTT packets to send, as instructed by the server.")
	@Expose
	@SerializedName("requested_num_packets")
	@JsonProperty(required = true, value = "requested_num_packets")
	private Integer requestedNumPackets;
	
	/**
	 * The actual number of sent RTT packets.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The actual number of sent RTT packets.")
	@JsonPropertyDescription("The actual number of sent RTT packets.")
	@Expose
	@SerializedName("num_sent")
	@JsonProperty(required = true, value = "num_sent")
	private Integer numSent;
	
	/**
	 * The actual number of received RTT packets.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The actual number of received RTT packets.")
	@JsonPropertyDescription("The actual number of received RTT packets.")
	@Expose
	@SerializedName("num_received")
	@JsonProperty(required = true, value = "num_received")
	private Integer numReceived;
	
	/**
	 * The actual number of failed RTT packets.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The actual number of failed RTT packets.")
	@JsonPropertyDescription("The actual number of failed RTT packets.")
	@Expose
	@SerializedName("num_error")
	@JsonProperty(required = true, value = "num_error")
	private Integer numError;
	
	/**
	 * The actual number of missing RTT packets.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The actual number of missing RTT packets.")
	@JsonPropertyDescription("The actual number of missing RTT packets.")
	@Expose
	@SerializedName("num_missing")
	@JsonProperty(required = true, value = "num_missing")
	private Integer numMissing;
	
	/**
	 * The actual size of RTT packets.
	 */
	@io.swagger.annotations.ApiModelProperty(required = true, value = "The actual size of RTT packets.")
	@JsonPropertyDescription("The actual size of RTT packets.")
	@Expose
	@SerializedName("packet_size")
	@JsonProperty(required = true, value = "packet_size")
	private Integer packetSize;

	public List<RttDto> getRtts() {
		return rtts;
	}

	public void setRtts(List<RttDto> rtts) {
		this.rtts = rtts;
	}

	public Integer getRequestedNumPackets() {
		return requestedNumPackets;
	}

	public void setRequestedNumPackets(Integer requestedNumPackets) {
		this.requestedNumPackets = requestedNumPackets;
	}

	public Integer getNumSent() {
		return numSent;
	}

	public void setNumSent(Integer numSent) {
		this.numSent = numSent;
	}

	public Integer getNumReceived() {
		return numReceived;
	}

	public void setNumReceived(Integer numReceived) {
		this.numReceived = numReceived;
	}

	public Integer getNumError() {
		return numError;
	}

	public void setNumError(Integer numError) {
		this.numError = numError;
	}

	public Integer getNumMissing() {
		return numMissing;
	}

	public void setNumMissing(Integer numMissing) {
		this.numMissing = numMissing;
	}

	public Integer getPacketSize() {
		return packetSize;
	}

	public void setPacketSize(Integer packetSize) {
		this.packetSize = packetSize;
	}
}

File: measurement-agent-swift/MeasurementAgentKit/Sources/DTO/RttInfoDto.swift
// MeasurementAgentKit: RttInfoDto.swift, created on 15.05.19
/*******************************************************************************
 * Copyright 2019 Benjamin Pucher (alladin-IT GmbH)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/

import Foundation

/// Contains information from a single round trip time measurement on the measurement agent.
public class RttDto: Codable {

    /// Round trip time recorded in nanoseconds.
    public var rttNs: UInt64?

    /// Relative time in nanoseconds (to test begin).
    public var relativeTimeNs: UInt64?

    public init() {

    }

    public init(rttNs: UInt64?, relativeTimeNs: UInt64?) {
        self.rttNs = rttNs
        self.relativeTimeNs = relativeTimeNs
    }

    ///
    enum CodingKeys: String, CodingKey {
        case rttNs = "rtt_ns"
        case relativeTimeNs = "relative_time_ns"
    }
}

/// Contains round trip time information measured during the measurement on the measurement agent.
public class RttInfoDto: Codable {

    /// List of all measured RTTs.
    public var rtts: [RttDto]?

    /// The number of RTT packets to send, as instructed by the server.
    public var requestedNumPackets: Int?

    /// The actual number of sent RTT packets.
    public var numSent: Int?

    /// The actual number of received RTT packets.
    public var numReceived: Int?

    /// The actual number of failed RTT packets.
    public var numError: Int?

    /// The actual number of missing RTT packets.
    public var numMissing: Int?

    /// The actual size of RTT packets.
    public var packetSize: Int?

    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    /// Average rtt in nanoseconds
    public var averageNs: UInt64?

    /// Maximum rtt in nanoseconds
    public var maximumNs: UInt64?

    /// Median rtt in nanoseconds
    public var medianNs: UInt64?

    /// Minimum rtt in nanoseconds
    public var minimumNs: UInt64?

    /// Standard deviation rtt in nanoseconds
    public var standardDeviationNs: UInt64?

    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    public init() {

    }

    ///
    enum CodingKeys: String, CodingKey {
        case rtts = "rtts"
        case requestedNumPackets = "requested_num_packets"
        case numSent = "num_sent"
        case numReceived = "num_received"
        case numError = "num_error"
        case numMissing = "num_missing"
        case packetSize = "packet_size"

        case averageNs = "average_ns"
        case maximumNs = "maximum_ns"
        case medianNs = "median_ns"
        case minimumNs = "minimum_ns"
        case standardDeviationNs = "standard_deviation_ns"
    }
}
